src/core/components/InstallButton/index.js

```diff
@@ -1,126 +1,46 @@
-        break;
-      case UNINSTALLING:
-        label = i18n.gettext('Uninstalling %(name)s.');
-        break;
-      case UNINSTALLED:
-        label = i18n.gettext('%(name)s is uninstalled. Click to install.');
-        break;
-      default:
-        label = i18n.gettext('Install state for %(name)s is unknown.');
-        break;
-    }
-    return i18n.sprintf(label, { name });
-  }
-
-  getDownloadProgress() {
-    const { downloadProgress, status } = this.props;
-    if (status === DOWNLOADING) {
-      return downloadProgress;
-    } else if ([INSTALLING, ENABLING].includes(status)) {
-      return Infinity;
-    } else if (status === UNINSTALLING) {
-      return -Infinity;
-    }
-    return undefined;
-  }
-
-  handleClick = (e) => {
-    e.preventDefault();
-    const {
-      guid, enable, install, installURL, name, status, installTheme, type, uninstall,
-    } = this.props;
-    if (type === THEME_TYPE && [UNINSTALLED, DISABLED].includes(status)) {
-      installTheme(this.themeData, guid, name);
-    } else if (status === UNINSTALLED) {
-      install();
-    } else if (status === DISABLED) {
-      enable();
-    } else if ([INSTALLED, ENABLED].includes(status)) {
-      uninstall({ guid, installURL, name, type });
-    }
+    installThemeAction: PropTypes.func,
   }
 
   render() {
-    const browsertheme = JSON.stringify(getThemeData(this.props));
-    const { handleChange, slug, status } = this.props;
-
-    if (!validStates.includes(status)) {
-      throw new Error(`Invalid add-on status ${status}`);
+    const { addon, hasAddonManager, i18n, installThemeAction } = this.props;
+    if (hasAddonManager) {
+      return <InstallSwitch {...this.props} />;
+    } else if (addon.type === THEME_TYPE) {
+      return (
+        <Button
+          data-browsertheme={JSON.stringify(getThemeData(addon))} onClick={installThemeAction}>
+          {i18n.gettext('Install Theme')}
+        </Button>
+      );
     }
-
-    const isChecked = [INSTALLED, INSTALLING, ENABLING, ENABLED].includes(status);
-    const isDisabled = status === UNKNOWN;
-    const isSuccess = [ENABLED, INSTALLED].includes(status);
-
-    return (
-      <div data-browsertheme={browsertheme} ref={(el) => { this.themeData = el; }}>
-        <Switch
-          checked={isChecked} disabled={isDisabled} progress={this.getDownloadProgress()}
-          name={slug} success={isSuccess} label={this.getLabel()}
-          onChange={handleChange} onClick={this.handleClick}
-          ref={(el) => { this.switchEl = el; }}
-        />
-      </div>
-    );
+    return <Button href={addon.installURL}>{i18n.gettext('Add to Firefox')}</Button>;
   }
 }
 
-export default translate()(InstallButtonBase);
+export function installTheme(event) {
+  event.preventDefault();
+  themeAction(event.currentTarget, THEME_INSTALL);
+}
+
+export const setInstallThemeAction = (Component) => (props) => (
+  <Component {...props} installThemeAction={installTheme} />
```

Comment:
This felt weird when I did it. I'm not sure why I did this. That seems reasonable.